src/py21cmfast/src/IonisationBox.c

```diff
@@ -115,19 +116,42 @@ LOG_SUPER_DEBUG("defined parameters");
 
     pixel_volume = pow(user_params->BOX_LEN/((float)(user_params->HII_DIM)), 3);
 
+    double alpha_esc_var = astro_params->ALPHA_ESC;
+    double norm_esc_var = astro_params->F_ESC10;
+    //TODO: cleanup the alpha/norm fits
+    if(flag_options->PHOTON_CONS_ALPHA){
+        //alpha_esc_var = get_alpha_fit(redshift);
+        norm_esc_var = get_alpha_fit(redshift);
+    }
 
-    if(flag_options->USE_MASS_DEPENDENT_ZETA) {
-        ION_EFF_FACTOR = global_params.Pop2_ion * astro_params->F_STAR10 * astro_params->F_ESC10;
+    //escape fractions taken into account in halo field
+    //TODO: control this in a different way
```

Comment:
There are a few points where different prefactors are assigned based on these three cases (halo field, mass-dependent zeta, fcoll) since the source field represents different quantities. It would be nice to consolidate these a little but its a minor issue to deal with when I try to modularise this file, similar to what was done in the spintemp code, I've removed the todo note for now and may make an issue later for the general reorganisation of this file.